rocAL/source/readers/image/numpy_data_reader.cpp

```diff
@@ -0,0 +1,587 @@
+/*
+Copyright (c) 2024 Advanced Micro Devices, Inc. All rights reserved.
+
+Permission is hereby granted, free of charge, to any person obtaining a copy
+of this software and associated documentation files (the "Software"), to deal
+in the Software without restriction, including without limitation the rights
+to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+copies of the Software, and to permit persons to whom the Software is
+furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in
+all copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.  IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+THE SOFTWARE.
+*/
+
+#include <math.h>
+
+#include <algorithm>
+#include <cassert>
+#include <cstring>
+#include <numeric>
+#include <random>
+
+#include "pipeline/commons.h"
+#include "pipeline/filesystem.h"
+#include "readers/image/numpy_data_reader.h"
+
+NumpyDataReader::NumpyDataReader() {
+    _src_dir = nullptr;
+    _sub_dir = nullptr;
+    _entity = nullptr;
+    _curr_file_idx = 0;
+    _current_file_size = 0;
+    _current_fPtr = nullptr;
+    _loop = false;
+    _shuffle = false;
+    _file_count_all_shards = 0;
+}
+
+unsigned NumpyDataReader::count_items() {
+    int size = get_max_size_of_shard(_batch_size, _loop);
+    int ret = (size - _read_counter);
+    if (_sharding_info.last_batch_policy == RocalBatchPolicy::DROP && _last_batch_padded_size != 0)
+        ret -= _batch_size;
+    return ((ret < 0) ? 0 : ret);
+}
```

Comment:
Check the count items in imag reader.cpp in other files we use the count items present in that

Reply:
@SundarRajan28 Is this done?